src/Data/Ord.purs

```diff
@@ -106,6 +105,9 @@ infixl 4 greaterThanOrEq as >=
 -- | Compares two values by mapping them to a type with an `Ord` instance.
 comparing :: forall a b. Ord b => (a -> b) -> (a -> a -> Ordering)
 comparing f = compare `on` f
+  where
```

Comment:
I think it might be a little nicer to just inline `on` here, instead of defining it and using it immediately.